tests/pthread_flood/main.c

```diff
@@ -43,56 +44,43 @@ int main(void)
     mutex_init(&testing_mutex);
     mutex_lock(&testing_mutex);
 
-
-    int pthread_cnt = 0;
-
-    pthread_t pthread_ids[MAXTHREADS];
-
     pthread_attr_t th_attr;
     pthread_attr_init(&th_attr);
 
-    if (-1 ==
-        pthread_create(&(pthread_ids[pthread_cnt % MAXTHREADS]), &th_attr, thread_func,
-                       NULL)) {
-        puts("[ERROR] cannot create pthreads");
-        return 0;
-    }
-    volatile int numthread_check = sched_num_threads;
-
-    int exit_loop  = -1;
+    int numthread_check = sched_num_threads;
 
-    puts("[START] Spawning threads");
-    do {
-        pthread_attr_setstackaddr(&th_attr,
-                                  &(dummy_stack[pthread_cnt % MAXTHREADS]));
+    puts("Spawning pthreads");
+    int pthread_cnt = 0;
+    for (uint8_t i = 0; i < MAXTHREADS; i++) {
```

Comment:
Here you could keep the do..while:

```c
do {
    /* create threads */
} while (++pthread_cnt < MAXTHREADS);
```

Reply:
The use of `++pthread_cnt` is important :)

Reply:
I would still need a break statement.

Reply:
It doesn't create `MAXTHREADS` but at most `MAXTHREADS`, it can run out of resources earlier than that, which is why the previous code checked the return code of `create`. Also why it used a module so at the end it will return an error for trying to use an already assigned pthread.

Reply:
Hmm I see, you are right.

Reply:
Can I trigger ci?